### PR TITLE
[BI-48] Export blog to pdf

### PR DESCRIPTION
**Pull request by alexstefan-binarcode**
The implementation of export blog to pdf functionality is made with the jsPDF library. The button can be found at the end of every blog post page.

Unfortunately, the generated PDF does not includes the image of the blog because of some CORS error.

Screenshot 2021-09-03 at 15 04 59

Screenshot 2021-09-06 at 10 01 43

### CODE REVIEW

File: src/views/Blog.vue [hunk 1]
```diff
@@ -57,12 +67,38 @@ export default {
...
+
     } catch (error) {
       this.notifyErrors(error);
     }
   },
   methods: {
     get,
+    jsPDF,
+    async getPDF() {
+      this.$refs.pdfContent.innerHTML = this.$refs.blog.innerHTML;
```
Comment (PascuMihai) [inline src/views/Blog.vue:79]: Is there a need for this assignation?
Can't use just use hte blog innerhtml and remove the hidden diff?

**Review by PascuMihai — APPROVED** [review]

File: src/views/Blog.vue [hunk 2]
```diff
@@ -74,6 +83,33 @@ export default {
...
   methods: {
     get,
+    jsPDF,
+
+    async getPDF() {
+      this.$refs.pdfContent.innerHTML = this.$refs.blog.innerHTML;
+      const page = this.$refs.pdfContent;
+      const filename = get(this.blog, 'attributes.title', 'blog');
+
+      var doc = new jsPDF({
```
Comment (cristijora) [inline src/views/Blog.vue:93]:
```suggestion
      const doc = new jsPDF({
```

File: src/views/Blog.vue [hunk 3]
```diff
@@ -74,6 +83,33 @@ export default {
...
+        orientation: "p",
+        format: "a4",
+        unit: "px"
+      });
+
+      await doc.html(page, {
+        html2canvas: {
+          scale: 0.38
+        },
+        callback: function (doc) {
```
Comment (cristijora) [inline src/views/Blog.vue:103]:
```suggestion
        callback: (doc) => {
```

File: src/views/Blog.vue [hunk 4]
```diff
@@ -74,6 +83,33 @@ export default {
   },
   methods: {
     get,
+    jsPDF,
+
+    async getPDF() {
+      this.$refs.pdfContent.innerHTML = this.$refs.blog.innerHTML;
+      const page = this.$refs.pdfContent;
```
Comment (cristijora) [inline src/views/Blog.vue:90]: This whole function could potentially fail due to many reasons
- Library configuration errors
- Missing refs or code related issues
Such methods should always have a `try/catch` block and in the `catch` notify the user about the error with a notification.

File: src/views/Blog.vue [hunk 5]
```diff
@@ -74,6 +83,33 @@ export default {
...
+      });
+
+      await doc.html(page, {
+        html2canvas: {
+          scale: 0.38
+        },
+        callback: function (doc) {
+          doc.save(`${filename}.pdf`);
+        },
+        x: 76.8,
```
Comment (cristijora) [inline src/views/Blog.vue:106]: Would suggest saving these into variables.
It's a bit unclear what these numbers mean. 
For example if we were to review this code in a couple of months, we would scratch our heads trying to understand what these numbers mean

**Review by Ceciliaa — APPROVED** [review]

**Pull request merged**